lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,216 @@
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: dellemc_idrac_firmware
+short_description: Firmware update from a repository on a network share (CIFS, NFS).
+version_added: "2.8"
+description:
+    - Update the Firmware by connecting to a network share (either CIFS or NFS) that contains a catalog of
+        available updates.
+    - Network share should contain a valid repository of Update Packages (DUPs) and a catalog file describing the DUPs.
+    - All applicable updates contained in the repository are applied to the system.
+    - This feature is available only with iDRAC Enterprise License.
+options:
+    idrac_ip:
+        required: True
+        description: iDRAC IP Address.
+    idrac_user:
+        required: True
+        description: iDRAC username.
+    idrac_pwd:
+        required: True
+        description: iDRAC user password.
+    idrac_port:
+        required: False
+        description: iDRAC port.
+        default: 443
+    share_name:
+        required: True
+        description: CIFS or NFS Network share.
+    share_user:
+        required: False
+        description: Network share user in the format 'user@domain' or 'domain\\user' if user is
+            part of a domain else 'user'. This option is mandatory for CIFS Network Share.
+    share_pwd:
+        required: False
+        description: Network share user password. This option is mandatory for CIFS Network Share.
+    share_mnt:
+        required: True
+        description: Local mount path of the network share with read-write permission for ansible user.
+            This option is mandatory for Network Share.
+    reboot:
+        required: False
+        description: Whether to reboots after applying the updates or not.
+        default: False
+        type: bool
+    job_wait:
+        required: False
+        description: Whether to wait for job completion or not.
+        type: bool
+        default: True
+    catalog_file_name:
+        required: False
+        description: Catalog file name relative to the I(share_name).
+        type: str
+        default: 'Catalog.xml'
+
+requirements:
+    - "omsdk"
+    - "python >= 2.7.5"
+author: "Rajeev Arakkal (@rajeevarakkal)"
+"""
+
+EXAMPLES = """
+---
+- name: Update firmware from repository on a Network Share
+  dellemc_idrac_firmware:
+       idrac_ip: "192.168.0.1"
+       idrac_user: "user_name"
+       idrac_pwd: "user_pwd"
+       share_name: "192.168.0.0:/share"
+       share_user: "share_user_name"
+       share_pwd: "share_user_pwd"
+       share_mnt: "/mnt/share"
+       reboot: True
+       job_wait: True
+       catalog_file_name: "Catalog.xml"
+"""
+
+RETURN = """
+---
+msg:
+  type: string
+  description: Over all firmware update status.
+  returned: always
+  sample: "Successfully updated the firmware."
+update_status:
+  type: dict
+  description: Firmware Update job and progress details from the iDRAC.
+  returned: success
+  sample:
+    sample: {
```

Comment:
1) you don't need another nested `sample` key as it's not what you return
2) please just use YAML for the structure description, it's a superset of JSON:
```suggestion
    InstanceID: JID_XXXXXXXXXXXX
    JobState: Completed
    Message: Job completed successfully.
    MessageId: REDXXX
    Name: Repository Update
    JobStartTime: NA
    Status: Success
```

Reply:
my bad..rectified